vpc/ec2wrapper/ec2metadata.go

```diff
@@ -174,6 +196,18 @@ func (mdc *EC2MetadataClientWrapper) getMetadata(path string) (string, error) {
 	return val, err
 }
 
+func ipStringToList(ips string) []string {
+	// This expects a non-\n terminated list of IPs separated by \n, and returns a normalized form
+	addresses := strings.Split(ips, "\n")
+	result := make([]string, len(addresses))
+	for idx, addr := range addresses {
+		sanitized := strings.Trim(strings.TrimSpace(addr), "\x00")
+		ip := net.ParseIP(sanitized)
```

Comment:
Worth checking if this returns `nil`, since this list comes from an untrusted source?  I mean, it's _probably_ fine...

Reply:
Fixed in next PR.